numpyro/distributions/directional.py

```diff
@@ -0,0 +1,120 @@
+import jax.numpy as np
+from jax import lax, random
+
+from numpyro.distributions import constraints
+from numpyro.distributions.distribution import Distribution
+from numpyro.distributions.util import validate_sample
+from numpyro.util import copy_docs_from
+
+
+@copy_docs_from(Distribution)
+class VonMises(Distribution):
+    arg_constraints = {'location': constraints.real, 'concentration': constraints.positive}
+    support = constraints.real
+    s_cutoff_map = {np.dtype(np.float16): 1.8e-1,
+                    np.dtype(np.float32): 2e-2,
+                    np.dtype(np.float64): 1.2e-4}
+
+    def __init__(self, location, concentration, validate_args=None):
+        # von Mises doesn't work for concentration=0, so set to tiny value
+        try:
+            dtype = np.dtype(concentration.dtype)
+        except AttributeError:
+            dtype = np.dtype(type(concentration))
+        concentration = np.maximum(concentration, np.finfo(dtype).tiny)
+        self.dtype = dtype
```

Comment:
Why do you need to store dtype? Something like `np.dtype(conc)` might be enough.

Reply:
`np.dtype(2.)` raise a type error, which I why stored to avoid repeating the try accept. I'll update it to: `dtype = dtypes.canonicalize_dtype(np.float64)`.